sequencer/src/block2.rs

```diff
@@ -139,17 +151,53 @@ impl QueryableBlock for BlockPayload {
         index: &Self::TransactionIndex,
     ) -> Option<(Self::Transaction, Self::InclusionProof)> {
         let vid = boilerplate::test_vid_factory(); // TODO temporary VID construction
-        let tx_range = self.get_tx_range(*index)?;
+
+        let (tx_range, tx_table_range_proof) = self.get_tx_range_with_proof(*index, &vid)?;
+
         Some((
             // TODO don't copy the tx bytes into the return value
             // https://github.com/EspressoSystems/hotshot-query-service/issues/267
             Transaction::new(crate::VmId(0), self.payload.get(tx_range.clone())?.to_vec()),
-            vid.payload_proof(&self.payload, tx_range).ok()?,
+            TxInclusionProof {
+                // TODO(795) every proof will contain a copy of the tx table len proof.
+                // Should we cache it in `BlockPayload` or something?
```

Comment:
👍

Reply:
Done in https://github.com/EspressoSystems/espresso-sequencer/pull/796/commits/be7e66b7483df7bec9f9be22128d29bc6cdf129d via `OnceLock`. That was a learning experience.